app/models/test_passage.rb

```diff
@@ -0,0 +1,51 @@
+class TestPassage < ApplicationRecord
+  belongs_to :user
+  belongs_to :test
+  belongs_to :current_question, class_name: 'Question', optional: true
+
+  before_validation :set_current_question
+
+  def completed?
+    current_question.nil?
+  end
+
+  def accept!(answer_ids)
+    self.correct_questions += 1 if correct_answer?(answer_ids)
+    save!
+  end
+
+  def passed?
+    questions = test.questions.count
+    total = correct_questions / questions.to_f * 100
+    total >= 85
+  end
+
+  def current_question_number
+    current_id = current_question.id
+    test.questions.where('id < ?', current_id).count + 1
```

Comment:
не настаиваю
но еще можно так:
```suggestion
    test.questions.where('id <= ?', current_question.id).count
```
и 24 строку убрать